cmd/openshift-install/create.go

```diff
@@ -267,6 +285,10 @@ func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args
 
 		err := runner(rootOpts.dir)
 		if err != nil {
+			if cmd.Name() == "cluster" && isClusterCreationError(err) {
```

Comment:
Can you look into adding something similar for install config validation errors?

Reply:
@patrickdillon I looked at the code a little more. And I am a little concerned about this. There are some errors that are obvious config related errors. But there are others I am not sure. For example, can I treat all errors returned from PreTerraform config errors? Or do I have to go into the function and differentiate different errors? Take an example of aws.PreTerraform, I can categorize PrivateSubnets and PublicSubnets as config errors. But how about errors from client interaction with AWS (such as add tags to subnet, or add tags to hosted zone)? Having to differentiate errors within those calls will be a nightmare. This also applies to other platforms like azure. 

The above is only for cluster asset. Many other assets are generated for “create cluster”: cluster.Metadata, machine.MasterIgnitionCustomizations, machine.WorkerIgnitionCustomizations, cluster.TerraformVariables, kubeconfig.AdminClient, password.KubeadminPassword, tls.JournalCertKey. Each of those have their own set of config errors. Can you clarify the ask is to cover all config errors?

Reply:
I just pushed a commit with some sample changes for config errors. This is not a complete list in any way. 

- I ignored marshing/unmarshing errors. 
- This change only applies to aws
- I have not considered some of the dependency assets

There should be a lot more places if this is the way to go. @patrickdillon I just want to confirm this is what you have in mind.

Reply:
@patrickdillon Thanks for pointing out the proper place for catching the validation errors. Now it looks much cleaner.